### PR TITLE
[tune+core] tune lifecycle and starting ray guide

### PR DESCRIPTION
**Pull request by richardliaw**
## Why are these changes needed?
Provide an indepth walkthrough of how tune's architecture works.

This helps users avoid issues of serialization/"why doesn't my code just work"?

I did not include a full diagram/walkthrough of schedulers/search algorithms and will do that in a later PR. 

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/tune/index.rst [hunk 1]
```diff
@@ -94,18 +94,18 @@ Reference Materials
 
 Here are some reference materials for Tune:
 
-  * :doc:`/tune/user-guide`
-  * :ref:`Frequently asked questions <tune-faq>`
-  * `Code <https://github.com/ray-project/ray/tree/master/python/ray/tune>`__: GitHub repository for Tune
+* :doc:`/tune/user-guide`
+* :ref:`Frequently asked questions <tune-faq>`
+* `Code <https://github.com/ray-project/ray/tree/master/python/ray/tune>`__: GitHub repository for Tune
```
Comment (richardliaw) [inline doc/source/tune/index.rst:99]: just removing extra indent

File: doc/source/tune/_tutorials/tune-lifecycle.rst [hunk 2]
```diff
@@ -0,0 +1,94 @@
+.. _tune-lifecycle:
```
Comment (richardliaw) [inline doc/source/tune/_tutorials/tune-lifecycle.rst:1]: this is the main document

File: doc/source/starting-ray.rst [hunk 3]
```diff
@@ -120,79 +104,67 @@ To stop or restart Ray, use the shutdown API.
...
 
-**Then on all of the other nodes**, run the following. Make sure to replace ``<address>`` with the value printed by the command on the head node (it should look something like ``123.45.67.89:6379``).
+Ray clusters can be launched with the :ref:`Cluster Launcher <ref-automatic-cluster>`.
+The ``ray up`` command uses the Ray cluster launcher to start a cluster on the cloud, creating a designated "head node" and worker nodes. Underneath the hood, it automatically calls ``ray start`` to create a Ray cluster.
 
-.. code-block:: bash
+Your code **only** needs to execute on one machine in the cluster (usually the head node). Read more about :ref:`running programs on a Ray cluster <using-ray-on-a-cluster>`.
 
-    $ ray start --address=<address>
+To connect to the existing cluster, similar to the method outlined in :ref:`start-ray-cli`, you must call ``ray.init`` and specify the address of the Ray cluster when initializing Ray in your code. This causes Ray to .
```
Comment (krfricke) [inline doc/source/starting-ray.rst:161]: There seem to be missing a half sentence at the end here

**Review by krfricke — CHANGES_REQUESTED** [review]
Minor correction request

File: doc/source/starting-ray.rst (hunk 3 above)
Comment (richardliaw) [reply doc/source/starting-ray.rst:161]:
```suggestion
To connect to the existing cluster, similar to the method outlined in :ref:`start-ray-cli`, you must call ``ray.init`` and specify the address of the Ray cluster when initializing Ray in your code. This allows Ray to connect to the cluster.
```

**Review by krfricke — APPROVED** [review]

**Pull request merged**